### PR TITLE
Jasan/bound sql

### PR DESCRIPTION
**Pull request by jsarenik**
Address #338

### CODE REVIEW

**Comment by rustyrussell** [comment]
What I'd really like is far more ambitious: a set of db wrappers which were used like:

```
   for_db(wallet->db, "SELECT x,y from Y WHERE foo=? and bar=?",
              DB_ARG(u64, &x),
              DB_ARG(struct pubkey, &y),
              DB_CONST_ARG(struct pubkey, &foo),
              DB_CONST_ARG(u64, bar))
        ...
```

This would set x and y for each row returned, and of course do compile time checks on the arguments and runtime checking that they matched the statement (both in number, and constness, perhaps in DEVELOPER mode with the type as well, though that's approximate).

and 
```
   alter_db(wallet->db, "DELETE FROM blocks WHERE hash = ?",
                 DB_ARG(struct sha256_double, &hash));
```

which is a simpler variant.

But it's terribly complex, which is why I haven't done it!

**Comment by jsarenik** [comment]
@rustyrussell I see your point, but why not adding these in the meantime in order to address the old issue? Or maybe rephrase #338 then.

The currently failing test on c75c5f1e is totally unrelated (`a2x` missing in one docker instance - new issue https://github.com/ElementsProject/lightning/issues/1642). So the tests are passing.
